src/components/search.tsx

```diff
@@ -11,14 +13,50 @@ export enum SearchType {
 }
 
 interface Props {
-  setSearch: (search: string) => void;
-  setSearchType: (searchType: SearchType) => void;
+  setSearch: (searchVal: SearchVal) => void;
 }
 
-export const Search: React.FC<Props> = ({ setSearch, setSearchType }) => {
-  const [val, setVal] = useState("");
+export const Search: React.FC<Props> = ({ setSearch }) => {
+  const [searchVal, setSearchVal] = React.useState({
+    search: "",
+    searchType: SearchType.Music,
+    resultsImgs: Array<Image>(),
+  });
   const [activeButton, setActiveButton] = useState("music");
 
+  useEffect(() => {
+    const download = async () => {
+      if (searchVal.search === "") {
+        return;
+      }
+
+      switch (searchVal.searchType) {
+        case SearchType.Games:
+          const albums = await getGame(searchVal.search);
+          setSearchVal({ ...searchVal, resultsImgs: albums });
+          break;
+
+        case SearchType.Movies:
+          const movies = await getMovie(searchVal.search);
+          setSearchVal({ ...searchVal, resultsImgs: movies });
+          break;
+
+        case SearchType.Series:
+          const series = await getSeries(searchVal.search);
+          setSearchVal({ ...searchVal, resultsImgs: series });
+          break;
+
+        default:
+          const games = await getAlbum(searchVal.search);
+          setSearchVal({ ...searchVal, resultsImgs: games });
+          break;
+      }
+    };
+
+    download();
+    setSearch(searchVal);
+  }, [searchVal, setSearch]);
```

Comment:
Any time `searchVal` is updated, this effect is triggered. That means that every time the search changes, it will fetch the images from the API, which is good, but it will also do the same each time it receives the images since they are part of `searchVal`! This ends up in an endless loop of requests to the API.